lib/runtime/src/ops/gather.cc

```diff
@@ -14,403 +14,181 @@
-      inputs[1]->data_type, index_domain, index_ptr);
-  out_of_memory = out_of_memory || (input_ptr == NULL) || (index_ptr == NULL);
-  Domain out_domain = sub_output.get_domain();
-  void *output_ptr =
-      sim->allocate(sub_output.get_volume(), outputs[0]->data_type);
-  out_of_memory = out_of_memory || (output_ptr == NULL);
-  cost_metrics.outputs_memory += cost_metrics.total_mem_diff_from(sim->offset);
-  GenericTensorAccessorW output_acc(
-      outputs[0]->data_type, out_domain, output_ptr);
-  if (out_of_memory) {
-    cost_metrics.forward_time = Simulator::MAXIMUM_TASK_RUN_TIME;
-    cost_metrics.backward_time = Simulator::MAXIMUM_TASK_RUN_TIME;
-    return true;
-  }
-
-  std::function<void()> forward, backward;
-  forward = [&] {
-    forward_kernel_wrapper(m, input_acc, index_acc, output_acc);
-  };
-  if (sim->computationMode == COMP_MODE_TRAINING) {
-    backward = [&] {
-      backward_kernel_wrapper(m, output_acc, index_acc, input_acc);
-    };
-  }
-
-  inner_measure_operator_cost(sim, forward, backward, cost_metrics);
-
-  if (sim->computationMode == COMP_MODE_TRAINING) {
-    printf("[Measure Gather] name(%s) forward_time(%.4lf) "
-           "backward_time(%.4lf)\n",
-           name,
-           cost_metrics.forward_time,
-           cost_metrics.backward_time);
-  } else {
-    printf("[Measure Gather] name(%s) forward_time(%.4lf)\n",
-           name,
-           cost_metrics.forward_time);
-  }
-  delete m;
-  return true;
+// template <task_id_t>
+// OpTaskSignature fwd_signature<GATHER_FWD_TASK_ID>() {
+//   OpTaskSignature fwd(OpTaskType::FWD);
+
+//   fwd.add_arg_slot<bool>(PROFILING);
+//   fwd.add_arg_slot<GatherAttrs>(ATTRS);
+
+//   fwd.add_input_slot(INPUT);
+//   fwd.add_output_slot(OUTPUT);
+//   fwd.add_weight_slot(INDEX);
+
+//   return fwd;
+// }
+
+// template <>
+// void register_task<GATHER_FWD_TASK_ID>() {
+//   register_task(GATHER_FWD_TASK_ID,
+//                 "Gather Fwd",
+//                 fwd_signature(),
+//                 forward_task);
+// }
+
+// template <task_id_t>
+// OpTaskSignature bwd_signature<GATHER_BWD_TASK_ID>() {
+//   OpTaskSignature bwd =
+//       infer_bwd_signature(get_op_signature(GATHER_FWD_TASK_ID));
+
+//   return bwd;
+// }
+
+// template <>
+// void register_task<GATHER_BWD_TASK_ID>() {
+//   register_task(GATHER_BWD_TASK_ID,
+//                 "Gather Bwd",
+//                 bwd_signature(),
+//                 backward_task);
+// }
+
+template <>
+void register_task<GATHER_FWD_TASK_ID>() {
+  OpTaskSignature fwd(OpTaskType::FWD);
+
+  fwd.add_arg_slot<bool>(PROFILING);
+  fwd.add_arg_slot<GatherAttrs>(ATTRS);
+
+  fwd.add_input_slot(INPUT);
+  fwd.add_output_slot(OUTPUT);
+  fwd.add_weight_slot(INDEX);
+  register_task(GATHER_FWD_TASK_ID, "Gather Fwd", fwd, forward_task);
 }
 
-}; // namespace FlexFlow
+template <>
+void register_task<GATHER_BWD_TASK_ID>() {
+  OpTaskSignature bwd =
+      infer_bwd_signature(get_op_signature(GATHER_BWD_TASK_ID));
```

Comment:
```suggestion
      infer_bwd_signature(get_fwd_signature(GATHER_BWD_TASK_ID));
```
And I think fix the `register_fwd_task`